cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
```

Comment:
Thanks for running the benchmark! Given that the time/op is so small (0.004ms vs 0.011ms), I think we can live with the performance impact of using a regexp. Could you add a comment in the code to signal to future readers that we actually paid attention to this? I’m thinking of something like “// Using a regexp so as to have less code than when using direct string processing. The performance impact is 0.011ms (regexp) as opposed to 0.004ms (string), which is acceptable.”